bpfprogs/probes.go

```diff
@@ -63,16 +74,19 @@ func (b *BPF) LoadBPFProgramProbeTypes(objSpec *ebpf.CollectionSpec) error {
 }
 
 // GetProgramSectionDetails returns group and name details
-// Section name format /prob type/group/name
+// Section name format type/prob type/group/name
+// ret : probe-type, group, probe-name
 // e.g.: tracepoint/sock/inet_sock_set_state
 // e.g.: kprobe/sys_execve
-func GetProgramSectionDetails(sectionName string) (string, string) {
+func GetProgramSectionDetails(sectionName string) (string, string, string) {
 	sections := strings.Split(sectionName, "/")
-	length := len(sections)
-	if length > 1 {
-		return sections[length-2], sections[length-1]
-	} else if length == 1 {
-		return "", sections[0]
+
+	switch strings.ToLower(sections[0]) {
+	case models.TracePoint:
+		return sections[0], sections[1], sections[2]
+	case models.KProbe, models.KRetProbe:
+		return sections[0], "", sections[1]
```

Comment:
Do we have control over the section name? It might be better to have it consistent if we can. It just seems weird that section 1 is sometimes a group and sometimes a name.

Reply:
Section name format SEC("<program_type>/<hook_name>/<optional_subtype>") -